src/ProposalTypesConfigurator.sol

```diff
@@ -64,38 +71,130 @@ contract ProposalTypesConfigurator is IProposalTypesConfigurator {
         return _proposalTypes[proposalTypeId];
     }
 
+    /**
+     * @notice Sets the scope for a given proposal type.
+     * @param proposalTypeId Id of the proposal type.
+     * @param txTypeHash A function selector that represent the type hash, i.e. keccak256("foobar(uint,address)").
+     * @param encodedLimit An ABI encoded string containing the function selector and relevant parameter values.
+     * @param parameters The list of byte represented values to be compared against the encoded limits.
+     * @param comparators List of enumuerated values represent which comparison to use when enforcing limit checks on parameters.
+     */
+    function setScopeForProposalType(
+        uint8 proposalTypeId,
+        bytes32 txTypeHash,
+        bytes calldata encodedLimit,
+        bytes[] memory parameters,
+        Comparators[] memory comparators
+    ) external override onlyAdmin {
+        _setScopeForProposalType(proposalTypeId, txTypeHash, encodedLimit, parameters, comparators);
+    }
+
+    /**
+     * @notice Sets the scope for a given proposal type.
+     * @param proposalTypeId Id of the proposal type.
+     * @param txTypeHash A function selector that represent the type hash, i.e. keccak256("foobar(uint,address)").
+     * @param encodedLimit An ABI encoded string containing the function selector and relevant parameter values.
+     * @param parameters The list of byte represented values to be compared against the encoded limits.
+     * @param comparators List of enumuerated values represent which comparison to use when enforcing limit checks on parameters.
+     */
+    function _setScopeForProposalType(
+        uint8 proposalTypeId,
+        bytes32 txTypeHash,
+        bytes calldata encodedLimit,
+        bytes[] memory parameters,
+        Comparators[] memory comparators
+    ) internal {
+        if (!_proposalTypesExists[proposalTypeId]) revert InvalidProposalType();
+        if (parameters.length != comparators.length) revert InvalidParameterConditions();
+
+        Scope memory scope = Scope(txTypeHash, encodedLimit, parameters, comparators);
+        scopes[proposalTypeId].push(scope);
+
+        scopeExists[proposalTypeId][txTypeHash] = true;
+
+        for (uint8 i = 0; i < _proposalTypes[proposalTypeId].txTypeHashes.length; i++) {
+            if (_proposalTypes[proposalTypeId].txTypeHashes[i] == txTypeHash) {
+                revert NoDuplicateTxTypes();
+            }
+        }
```

Comment:
move to revert as early as possible (so it would be after checking params are valid)